CRM/Activity/BAO/Activity.php

```diff
@@ -2107,6 +2107,7 @@ public static function addActivity(
     $id = CRM_Core_Session::getLoggedInContactID();
     if ($id) {
       $activityParams['source_contact_id'] = $id;
+      $activityParams['target_contact_id'][] = $activity->contact_id;
```

Comment:
Can you explain the logic here? I worry there might be cases where it is not populated..